database/seeders/StudentTagsSeeder.php

```diff
@@ -0,0 +1,123 @@
+        "Pain" => [ "migraine", "pain"], 
+        "Personality Psychology" => [ "narcissism", "personality", "prosociality", "behavior", "personality pathology"], 
+        "Social Psychology" => ["body image", "body perception", "couples", "cross-cultural", "cultural diversity", "culture", "emotion", "emotions", "family", "interaction", "intimacy", "morality", "relationships", "signaling", "social communication",  "social trait perception",  "socioeconomic status"], 
+        "Speech / Language" => ["bilingual", "communication", "developmental language disorder", "language", "language development", "literacy", "motor", "semantic", "speech", "speech development", "speech perception", "speech therapy", "word learning"], 
+        "Speech and Language Disorders" => ["assessment", "disordered", "SLI"], 
+        "Statistics / Measurement / Modeling / Machine Learning" => ["artificial intelligence", "computational", "computational modeling", "dyadic data analysis", "eyetracking", "face recognition", "machine learning", "mathematical psychology", "psychometrics"], 
+    ];
+    
+    /**
+     * Run the StudentTags Seeder
+     *
+     * @return void
+     */
+    public function run()
+    {
+
+        $this->createNewTags();
+        $this->syncStudentApplicationsNewTags();
+
+    }
+
+    /**
+     * Create new tags with the type "App\StudentNew"
+     *
+     * @return void
+     */
+    public function createNewTags()
+    {
+        Tag::findOrCreate(collect($this::NEW_TAGS_ASSOCIATION)->keys(), "App\StudentNew");
+    }
+    /**
+     * Loop through the students with tags,
+     * find the new tags associated to store them in an array
+     * and sync the new tags for each student.
+     *
+     * @return void
+     */
+    public function syncStudentApplicationsNewTags()
+    {   
+        
+        $new_tags_association = collect($this::NEW_TAGS_ASSOCIATION);
+        $new_tags = Tag::WithType("App\StudentNew")->get();
+        $students_with_tags = Student::with('tags')->has('tags')->get();
+
+        foreach ($students_with_tags as $student) {
+
+            $student_new_tags = [];
+            $tags_not_found = [];
+            
+            foreach ($student->tags as $student_old_tag) {
+
+                $elem = $new_tags_association->filter(function ($value, $key) use ($student_old_tag) {
+                    return in_array($student_old_tag->name, $value);
+                });
+
+                $new_tag = $new_tags->filter(function($item) use ($elem) {
+                    return trim(strtolower($item->name)) == trim(strtolower($elem->keys()->first()));
+                })->first();
+
+                if (isset($new_tag)) {
```

Comment:
It would be good to also only add the `$new_tag` to `$student_new_tags` if it isn't already there, so we're not over-counting when we sync and log them. So, maybe something like `if (isset($new_tag) && !in_array($new_tag, $student_new_tags))`